### PR TITLE
[API-15184] - ClaimsApi - ITF POST permit 'claimant_ssn' attribute

### PR DESCRIPTION
**Pull request by jw81**
## Description of change
- Adds support for new `claimant_ssn` field on POST ITF submissions
- Adds `claimant_ssn` description to dev docs
- Provides better description for `veteran_participant_id` field in dev docs
- Provides better description for `claimant_participant_id` field in dev docs

## Original issue(s)
[API-15184](https://vajira.max.gov/browse/API-15184)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -41,7 +41,7 @@ gem 'aws-sdk-kms'
 gem 'aws-sdk-s3', '~> 1'
 gem 'aws-sdk-sns', '~> 1'
 gem 'betamocks', git: 'https://github.com/department-of-veterans-affairs/betamocks', branch: 'master'
-gem 'bgs_ext', git: 'https://github.com/department-of-veterans-affairs/bgs-ext.git', require: 'bgs'
+gem 'bgs_ext', git: 'https://github.com/department-of-veterans-affairs/bgs-ext.git', branch: 'API-15184-ClaimsApi-insertintenttofile-allow-optional-clmantSsn-attribute', require: 'bgs'
```
Comment (hinzed1127) [inline Gemfile:44]: You'll wanna revert this once https://github.com/department-of-veterans-affairs/bgs-ext/pull/74 gets merged

File: Gemfile.lock [hunk 2]
```diff
@@ -42,7 +42,8 @@ GIT
 
 GIT
   remote: https://github.com/department-of-veterans-affairs/bgs-ext.git
-  revision: 4e216f8097fe93c4ca76d06f358c8d169afcd7ad
+  revision: ec28b2c269444fd42d9851a9c600c5eecd24552e
+  branch: API-15184-ClaimsApi-insertintenttofile-allow-optional-clmantSsn-attribute
```
Comment (hinzed1127) [inline Gemfile.lock:46]: same as above

File: modules/claims_api/app/controllers/claims_api/v1/forms/intent_to_file_controller.rb [hunk 3]
```diff
@@ -70,14 +70,31 @@ def validate
...
+        def handle_claimant_fields(options:, form_attributes:, target_veteran:)
+          claimant_ssn = form_attributes['claimant_ssn']
+          participant_claimant_id = form_attributes['participant_claimant_id']
+
+          options[:claimant_ssn] = claimant_ssn if claimant_ssn
+          options[:participant_claimant_id] = participant_claimant_id if participant_claimant_id
+
+          # if neither field was provided, then default to sending 'participant_claimant_id'
+          if options[:claimant_ssn].blank? && options[:participant_claimant_id].blank?
+            options[:participant_claimant_id] = target_veteran.participant_id
```
Comment (hinzed1127) [inline modules/claims_api/app/controllers/claims_api/v1/forms/intent_to_file_controller.rb:94]: is this a default that we wanna call out in documentation at all?

**Review by hinzed1127 — CHANGES_REQUESTED** [review]
Need to revert Gemfile changes and wait for https://github.com/department-of-veterans-affairs/bgs-ext/pull/74 to get merged first

**Review by acovrig — DISMISSED** [review]

File: Gemfile (hunk 1 above)
Comment (jw81) [reply Gemfile]: Updated [here](https://github.com/department-of-veterans-affairs/vets-api/pull/9810/commits/a1dee3955147e8d42b26c8c651c479c5712c5566).

File: Gemfile.lock (hunk 2 above)
Comment (jw81) [reply Gemfile.lock]: Updated [here](https://github.com/department-of-veterans-affairs/vets-api/pull/9810/commits/a1dee3955147e8d42b26c8c651c479c5712c5566).

**Review by hinzed1127 — APPROVED** [review]

**Pull request merged**